spec/controllers/account/admin/companies_controller_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe Account::Admin::CompaniesController, type: :controller do
+  let!(:company) { create(:company) }
+  let!(:valid_params) { attributes_for :company }
+  let!(:invalid_params) { { name: ' ' } }
+
+  describe 'GET#index' do
+    login_user
+    it 'assigns companies and renders template' do
+      get :index
+      expect(response).to have_http_status(:success)
+      expect(assigns(:companies)).to eq([company])
+      expect(response).to render_template('index')
+    end
+  end
+
+  describe 'GET#show' do
+    login_user
+    before do
+      get :show, params: { id: company.id }
+    end
+
+    it 'returns success and assigns company' do
+      expect(response).to have_http_status(:success)
+      expect(assigns(:company)).to eq(company)
+    end
+  end
+
+  describe 'GET#new' do
+    login_user
+    it 'returns success and assigns company' do
+      get :new
+      expect(response).to have_http_status(:success)
+      expect(assigns(:company)).to be_a_new(Company)
+    end
+  end
+
+  describe 'POST#create' do
+    login_user
+    context 'with valid params' do
+      it 'creates a new company' do
+        expect do
+          post :create, params: { company: valid_params }
+        end.to change(Company, :count).by(1)
+      end
+
+      it 'redirects to the created company' do
+        post :create, params: { company: valid_params }
+        expect(response).to have_http_status(:redirect)
+        expect(response).to redirect_to(account_admin_company_path(Company.last))
+      end
+    end
+
+    context 'with invalid params' do
+      it 'do not create a new company' do
+        expect do
+          post :create, params: { company: invalid_params }
+        end.not_to change(Company, :count)
+      end
+    end
+  end
+
+  describe 'GET#edit' do
+    login_user
+    before do
+      get :edit, params: { id: company.id }
+    end
+
+    it 'returns http success and assign company' do
+      expect(response).to have_http_status(:success)
+      expect(assigns(:company)).to eq(company)
+    end
+  end
+
+  describe 'PUT#update' do\
+    login_user
+    context 'with valid params' do
+      before do
+        put :update, params: { id: company.id,
+                               company: valid_params.merge!(name: 'Example',
+                                                            phone: '0123456789',
+                                                            email: 'example@email.com') }
+      end
+
+      it 'assigns the company' do
```

Comment:
it { expect(assigns(:company)).to eq(company)
it { expect(response).to have_http_status(:redirect) }
it { expect(response).to redirect_to.....
EVERY WERE